test/pending_log_event_test.exs

```diff
@@ -0,0 +1,33 @@
+defmodule LogflareLogger.PendingLoggerEventTest do
+  use ExUnit.Case
+  alias LogflareLogger.PendingLoggerEvent
+
+  describe "changeset/2" do
+    test "list with different types handled in metadata" do
+      body = %{
+        "metadata" => %{
```

Comment:
we can simplify this to
```
body = %{"my_list" => [ 123, 1, "123" ]}
... # assertions here


body = %{"my_list" => [ 123, [1] ]}
.... # assertions

body = %{"my_list" => [ 123, %{"a"=> 1} ]}
.... # assertions
```
and we can do a pattern match on the changeset changes instead, which is less verbose.

Reply:
also missing a test case for list of maps.

Reply:
Wanted to have the error closer to what was happening but can do 👍

Reply:
kept the use case closer to the error
added a list of maps to check if nothing is changed in those scenarios